src/core/models.py

```diff
@@ -640,6 +712,16 @@ def node_ip(self) -> str:
         """
         return self.k8s.get_node_ip(self.pod_name)
 
+    @property
+    def directory_id(self) -> str:
+        """Directory ID of the node as saved in `meta.properties`."""
+        return self.relation_data.get("directory-id", "")
+
+    @property
+    def added_to_quorum(self) -> bool:
```

Comment:
A couple of important points here, not really blocking but I think they are good for future improvements
- Logic around `added_to_quorum` should probably be checked against the cluster itself, and be tracked as little as possible on the charm.
- Something like an upgrade or a failure on the unit will not be correctly resolved. 
At the moment there is no _unsetting_ of this property, it will stay as `true` once the unit is first added to the quorum, so a failing unit would look at the databag, see `added_to_quorum=True` when the reality is that the unit is on a recovery state.